src/ui/api/search.go

```diff
@@ -39,33 +40,25 @@ type searchResult struct {
 
 // Get ...
 func (s *SearchAPI) Get() {
-	userID, _, ok := s.GetUserIDForRequest()
-	if !ok {
-		userID = dao.NonExistUserID
-	}
-
 	keyword := s.GetString("q")
-
-	isSysAdmin, err := dao.IsAdminRole(userID)
-	if err != nil {
-		log.Errorf("failed to check whether the user %d is system admin: %v", userID, err)
-		s.CustomAbort(http.StatusInternalServerError, "internal error")
-	}
+	isAuthenticated := s.SecurityCtx.IsAuthenticated()
+	username := s.SecurityCtx.GetUsername()
+	isSysAdmin := s.SecurityCtx.IsSysAdmin()
 
 	var projects []*models.Project
+	var err error
 
-	if isSysAdmin {
-		projects, err = dao.GetProjects(nil)
-		if err != nil {
-			log.Errorf("failed to get all projects: %v", err)
-			s.CustomAbort(http.StatusInternalServerError, "internal error")
-		}
+	if !isAuthenticated {
```

Comment:
if !isAuthenticated {
	projects, err = s.ProjectMgr.GetPublic()
} 
if isSysAdmin {
	projects, err = s.ProjectMgr.GetAll(nil)
} else {
	projects, err = s.ProjectMgr.GetHasReadPerm(username)
}

It could be better to understand.